c++/src/kj/string.h

```diff
@@ -451,6 +548,53 @@ auto toCharSequence(T&& value) -> decltype(_::STR * kj::fwd<T>(value)) {
 }
 
 namespace _ {  // private
+template <typename T, typename = void>
+struct IsStringifiedToSignalSafeCharSequence_ {
+  // This is the base implementation for non-stringifiable types (KJ_STRINGIFY not implemented).
```

Comment:
Won't such types fail to compile anyway? Why spend the effort to handle them here?

Reply:
The issue is described below:
```
  // This double-dispatch is needed so that ArrayPtr<T> works correctly. Otherwise the SFINAE
  // check for T being stringifiable would be ambiguous in some cases due to the implicit handling
  // of wrapped array types below.
```
Open to suggestions. Getting all this SFINAE building & working correctly is challenging so I'm not keen to do exploratory work unless there's a concrete suggestion that's been tested. I've lost too many hours finagling this one way & then doing it again for the rewrite.

Reply:
1. I don't think that's what I was asking. I asked why we need a specialization to handle non-stringifiable types. That doesn't appear to have anything to do with the ArrayPtr ambiguity?
2. My suggestion, per my next comment, would be not to special-case any types here, especially given that the special-cases here appear to be inconsistent with the actual implementations.
3. I'm not keen to merge code that has unexplained redundancies or inconsistencies, as those tend to lead to maintenance burden and bugs down the road. If you don't feel it's worth the effort to get it right then we should just close this PR.

Reply:
`isStringifiedToSignalSafeCharSequence()` is defined as returning false is the type isn't stringifiable. This is a valid situation that occurs, for example, in `DebugComparison`.